src/lib.rs

```diff
@@ -11,34 +13,37 @@ pub struct ScuttleButt {
 }
 
 impl ScuttleButt {
-
     pub fn with_node_id(self_node_id: String) -> Self {
         ScuttleButt {
             max_transmitted_key_values: 10,
             self_node_id,
-            cluster_state_map: ClusterState::default()
+            cluster_state_map: ClusterState::default(),
         }
     }
 
     pub fn set_max_num_key_values(&mut self, max_transmitted_key_values: usize) {
         self.max_transmitted_key_values = max_transmitted_key_values;
     }
 
-    pub fn create_syn_message(&mut self) -> ScuttleButtMessage {
+    pub fn create_syn_message(&self) -> ScuttleButtMessage {
```

Comment:
I've made some minor unrelated changes in this file simply to satisfy clippy/rustfmt. Could revert them if desired but I think improving code quality is always nice.

Reply:
absolutely.